docs/source/contributing/developing.adoc

```diff
@@ -274,12 +274,26 @@ RUN_BEFORE_FEATURE::
 The `RUN_BEFORE_FEATURE` parameter specifies {project} commands to be run before each feature.
 This provides the ability to run integration tests against a {project} instance which is not in its default state.
 When multiple commands are specified, they must be delimited by a semicolon.
-For example, tests can be run against a stopped {project} instance with the image caching option turned on by running the following:
+For example, tests can be run against a stopped {project} instance with the image caching option turned on by running the following: 
 
 ----
 $ make integration_all RUN_BEFORE_FEATURE="start; stop; config set image-caching true"
 ----
 
+TEST_WITH_SHELL::
+
+About:::
+Majority of integration tests in Minishift passes command arguments directly to `minishift` or `oc` binaries with usage of Go's `os/exec` package.
+However some integration tests first starts an instance of shell, into which the tests later passes the commands.
+This attitude allows to test `minishift docker-env` or `minishift oc-env` commands, which are dependent on which kind of shell they are being executed in.
```

Comment:
```adoc
This allows testing of commands which are dependent on the shell they are being executed in.
```

I don't think we need to explicitly state which commands are dependent on the shell. If that's necessary, though, I would add this information to the end of the sentence in a parenthetical:

```(such as `minishift docker-env` and `minishift oc-env`)```

Reply:
Will keep the shorter variant without mentioning the exact commands, thank you.